### PR TITLE
Implement script for fixing copyright dates, adjust copyright to 2022

### PR DESCRIPTION
**Pull request by dreamer**
- Implement script to be used once a year ;)
- Adjust several copyright and SPDX statements in files introduced/heavily modified lately
- In the process I noticed several small files encoded with windows line endings, so I decided to fix it now
- Commit updated copyright lines

*If other maintainers want to, I can update their copyright lines to 2022 as well.*

### CODE REVIEW

File: include/ansi_code_markup.h [hunk 1]
```diff
@@ -1,5 +1,7 @@
 /*
- *  Copyright (C) 2022  The DOSBox Staging Team
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2022-2022  The DOSBox Staging Team
```
Comment (shermp) [inline include/ansi_code_markup.h:4]: Probably should remain `2022`

File: src/dos/dos_resources.h [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- *  Copyright (C) 2022  The DOSBox Staging Team
+ *  Copyright (C) 2022-2022  The DOSBox Staging Team
```
Comment (shermp) [inline src/dos/dos_resources.h:2]: Just 2022

File: src/misc/ansi_code_markup.cpp [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /*
  *  SPDX-License-Identifier: GPL-2.0-or-later
  *
- *  Copyright (C) 2022  The DOSBox Staging Team
+ *  Copyright (C) 2022-2022  The DOSBox Staging Team
```
Comment (shermp) [inline src/misc/ansi_code_markup.cpp:4]: Just 2022

File: tests/ansi_code_markup_tests.cpp [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /*
  *  SPDX-License-Identifier: GPL-2.0-or-later
  *
- *  Copyright (C) 2022  The DOSBox Staging Team
+ *  Copyright (C) 2022-2022  The DOSBox Staging Team
```
Comment (shermp) [inline tests/ansi_code_markup_tests.cpp:4]: Just 2022

**Review by shermp — COMMENTED** [review]
Just a few `2022-2022` instances I noticed.

File: include/ansi_code_markup.h (hunk 1 above)
Comment (dreamer) [reply include/ansi_code_markup.h]: Result of me running this script several times :) Thanks for spotting this!

I added `git grep` invocation to the script to highlight such changes to the person updating copyright lines. Also, several files already had "2022-2022" before - I changed them to have a single year instead.

**Comment by kcgen** [comment]
Thanks, @dreamer.

Regarding copyright dates, I'd prefer the copyright years start with the original work and end with the latest revision year.

US copyright (in which the GPL's terms are legally binding) [says that](https://www.copyright.gov/help/faq/faq-duration.html) the copyright duration for pseudonymous works endures for a term of 95 years from the year of its first publication or a term of 120 years from the year of its creation, whichever expires first.

Minor revisions (typos, spelling, grammar, etc) are considered part of the original release, and so the original copyright terms remains in place.

Substantial revisions [are considered derived works](https://www.copyright.gov/circs/circ14.pdf
) that warrant a second copyright covering the altered material. The second copyright doesn't enlarge the scope of the original, but only protects the new material.

If I released a book in 2000 and was still selling it today, the material would still be copyright 2000 (not 2022).  The only way I could stamp it with a 2022 date would be to substantially revise it -- and even then, only the revised portions would carry the new 2022 copyright. 

Based on these points, the copyright dates on a source file containing both original and substantially revised content should include the original date and the revision dates; or more loosely, the last revision date.

So I'd prefer we dial-back our policy to match the above, and only stamp new dates on files when they've been notably changed.  Yes.. it's a lot more work to get this right, and surely error prone. But I'd rather not stamp new dates on files carte blanche anymore.

Is it possible to update your script to inspect the last commit dates to a file? (and ideally, it could even filter our these carte-blanche copyright date commits?)

For example, `opl.h` was created by the DOSBox Team in 2009 and was last updated in 2009, however since then, it's copyright date has been updated:

``` text
commit 1b6028a8be78b3a79e4909eff64f7f57770350e9
Author: Sebastian Strohhäcker <c2woody@users.sourceforge.net>
Date:   Fri Apr 17 17:24:47 2009 +0000

    add alternative opl2/opl3 emulator; fix opl clock rate
    
    
    Imported-from: https://svn.code.sf.net/p/dosbox/code-0/dosbox/trunk@3348

commit 05da192ef7747b33887fbb256cf41ccc805987cf
Author: Sebastian Strohhäcker <c2woody@users.sourceforge.net>
Date:   Fri May 15 21:07:13 2009 +0000

    small cleanup
    
    
    Imported-from: https://svn.code.sf.net/p/dosbox/code-0/dosbox/trunk@3390

commit a78b2c4f8b87c89ab7dbf1d53f34b276f77d098e
Author: Peter Veenstra <qbix79@users.sourceforge.net>
Date:   Thu Feb 11 13:36:50 2010 +0000

    Update year
    
    Imported-from: https://svn.code.sf.net/p/dosbox/code-0/dosbox/trunk@3548

commit 999f33de27f8285c1f31bb17ce564fccdeabe1c3
Author: Peter Veenstra <qbix79@users.sourceforge.net>
Date:   Tue Apr 26 09:34:55 2011 +0000

    Update year. Remove CVS tags
    
    Imported-from: https://svn.code.sf.net/p/dosbox/code-0/dosbox/trunk@3700

commit 2fea508f7fd8f5ce2e3cc33a902603f336e5805c
Author: Peter Veenstra <qbix79@users.sourceforge.net>
Date:   Tue Jan 15 09:03:13 2013 +0000

    Year update
    
    Imported-from: https://svn.code.sf.net/p/dosbox/code-0/dosbox/trunk@3812

commit 0f4c92ca3267dc2bfb0cdca5491e691e99954eb0
Author: Peter Veenstra <qbix79@users.sourceforge.net>
Date:   Tue Jan 6 14:40:32 2015 +0000

    Year update.
    
    Imported-from: https://svn.code.sf.net/p/dosbox/code-0/dosbox/trunk@3878

commit 6cf39c1fc614e0232fc1629179cfa4603098ed38
Author: Peter Veenstra <qbix79@users.sourceforge.net>
Date:   Tue May 30 11:35:08 2017 +0000

    Years update
    
    Imported-from: https://svn.code.sf.net/p/dosbox/code-0/dosbox/trunk@4021

commit 5bb10db88bb32bea54aef17836a61e4a04d226a5
Author: Peter Veenstra <qbix79@users.sourceforge.net>
Date:   Tue May 29 12:58:58 2018 +0000

    Time keeps on ticking
    
    Imported-from: https://svn.code.sf.net/p/dosbox/code-0/dosbox/trunk@4109

commit 1fbaff47fc54267a865790048213bcc10abfb17c
Author: Peter Veenstra <qbix79@users.sourceforge.net>
Date:   Fri Jan 25 14:09:58 2019 +0000

    Update year and address of FSF
    
    Imported-from: https://svn.code.sf.net/p/dosbox/code-0/dosbox/trunk@4183
```

**Comment by dreamer** [comment]
@kcgen I updated the script to fix copyright only in:

- files with commits this year (without deeper inspection to filter-out copyright bumps from SVN)
- metadata files and "--version" message (these inform about copyright of whole package or binary, therefore should be retouched regularly before each release)

The only real consequence I can think of is that now we must to run this script twice a year:

- before year ends (to catch all files that were touched but author forgot to update copyright line)
- early in the year (to update metadata and prepare for any releases in incoming year)

**Comment by kcgen** [comment]
Looks great @dreamer :+1: 

Added notes on bumping the copyright dates here:

https://github.com/dosbox-staging/dosbox-staging/wiki/Maintenance-and-Release-Notes#52-update-copyright-dates

**Pull request merged**